### ISSUE TITLE
Question about AdaIN

### ISSUE THREAD

**Issue by fxcdl** [jonrei/tf-AdaIN#10]

**Comment by fxcdl** [comment]
Hi
In AdaIN.py line 123,could you tell me why are you using ‘style_mean’ and ‘tf.sqrt(style_variance)’  in the fuction ‘tf.nn.batchnorm'，and in the line 126，why are you adding ‘(1 - alpha) * content_features’ at the end of the formula，In the original text, the author uses  'a(y) *（x-mean）/variance + u(y)', did you mean that‘(1 - alpha) * content_features’ is  'a(y) *（x-mean）/variance + u(y)'？

**Comment by jonrei** [comment]
Hi fxcdl,

The tensorflow batch_normalization is an implementation of Eq.1. If you look at Eq. 8, it is quite similar and it is possible to reuse the batch_normalization function to calculate the AdaIN equation by substituting β (offset) and 𝛾 (scale) with  μ (`style_mean`) and σ (`tf.sqrt(style_variance)`).

`alpha` is a scaling factor telling how much of the original image's style should be kept, and how much we should try to take from the style image.

**Comment by fxcdl** [comment]
I understand！Thank you very much for your reply!

**Comment by Carrotor116** [comment]
In paper, `AdaIN(x,y) = sigma(y)*((x-mu(x))/sigma(x))+mu(x)`  is described as Eq. 8. 
And in [tf API document](https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization), `tf.nn.batch_normalization` is described as follow :
```python
tf.nn.batch_normalization(
    x,
    mean,
    variance,
    offset,
    scale,
    variance_epsilon,
    name=None
)
```
> `scale`: A scale Tensor, often denoted gamma in equations, or `None`. If present, the scale is applied to the normalized tensor.

So I am confused why you use `tf.sqrt(style_variance)` as `scale` install of `style_variance` ?

**Comment by fxcdl** [comment]
> In paper, `AdaIN(x,y) = sigma(y)*((x-mu(x))/sigma(x))+mu(x)` is described as Eq. 8.
> And in [tf API document](https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization), `tf.nn.batch_normalization` is described as follow :
> 
> ```python
> tf.nn.batch_normalization(
>     x,
>     mean,
>     variance,
>     offset,
>     scale,
>     variance_epsilon,
>     name=None
> )
> ```
> 
> > `scale`: A scale Tensor, often denoted gamma in equations, or `None`. If present, the scale is applied to the normalized tensor.
> 
> So I am confused why you use `tf.sqrt(style_variance)` as `scale` install of `style_variance` ?

sigma(y) means standard deviation of y,but 'style_variance' is variance

**Comment by Carrotor116** [comment]
> > In paper, `AdaIN(x,y) = sigma(y)*((x-mu(x))/sigma(x))+mu(x)` is described as Eq. 8.
> > And in [tf API document](https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization), `tf.nn.batch_normalization` is described as follow :
> > ```python
> > tf.nn.batch_normalization(
> >     x,
> >     mean,
> >     variance,
> >     offset,
> >     scale,
> >     variance_epsilon,
> >     name=None
> > )
> > ```
> > 
> > 
> > > `scale`: A scale Tensor, often denoted gamma in equations, or `None`. If present, the scale is applied to the normalized tensor.
> > 
> > 
> > So I am confused why you use `tf.sqrt(style_variance)` as `scale` install of `style_variance` ?
> 
> sigma(y) means standard deviation of y,but 'style_variance' is variance



oh, the gamma(x) in formulation of bn is the standard deviation of x. I understood.
thanks for your reply
